src/openms_gui/source/VISUAL/SpectraIdentificationViewWidget.cpp

```diff
@@ -1059,65 +585,68 @@ namespace OpenMS
   }
 
   // Upon changes in the table data (only possible by checking or unchecking a checkbox right now),
-  // update the corresponding PeptideIdentification / PeptideHits
-  void SpectraIdentificationViewWidget::updateData_(QTableWidgetItem* item)
+  // update the corresponding PeptideIdentification / PeptideHits by adding a metavalue: 'selected'
+  void SpectraIdentificationViewWidget::updatedSingleCell_(QTableWidgetItem* item)
   {
     // no valid peak layer attached
     if (layer_ == nullptr || layer_->getPeakData()->size() == 0 || layer_->type != LayerData::DT_PEAK)
     {
       return;
     }
 
-    // Find column indices by names
-    Size n_col = table_widget_->columnCount();
-    Size id_col = 0;
-    Size ph_col = 0;
-
-    for (Size c = 0; c < n_col; ++c)
-    {
-      String col_head = table_widget_->horizontalHeaderItem(c)->text();
-      if (col_head == "#ID")
-      {
-        id_col = c;
-      }
-      if (col_head == "#PH")
-      {
-        ph_col = c;
-      }
-    }
-
     // extract position of the correct Spectrum, PeptideIdentification and PeptideHit from the table
-    int r = item->row();
-    bool selected = item->checkState() == Qt::Checked;
-    int spectrum_index = table_widget_->item(r, 1)->data(Qt::DisplayRole).toInt();
-    int num_id = table_widget_->item(r, id_col)->text().toInt();
-    int num_ph = table_widget_->item(r, ph_col)->text().toInt();
+    int row = item->row();
+    String selected = item->checkState() == Qt::Checked ? "true" : "false";
```

Comment:
Why String now? And shouldn't this be a property of the row and not the PeptideHit associated to that row? What if the rows represent sth else?

Reply:
it was string before. I just removed the intermediate step.
The row represents a single PH and it cannot represent sth else (except when we show unidentified MS2 spectra, but they do not have an editable checkbox, so this functions never gets called for those rows).